configs/platforms/osx-10.14-x86_64.rb

```diff
@@ -0,0 +1,23 @@
+platform 'osx-10.14-x86_64' do |plat|
+    plat.servicetype 'launchd'
+    plat.servicedir '/Library/LaunchDaemons'
+    plat.codename "mojave"
+    plat.provision_with 'export HOMEBREW_NO_EMOJI=true'
+    plat.provision_with 'export HOMEBREW_VERBOSE=true'
+    plat.provision_with 'sudo dscl . -create /Users/test'
+    plat.provision_with 'sudo dscl . -create /Users/test UserShell /bin/bash'
+    plat.provision_with 'sudo dscl . -create /Users/test UniqueID 1001'
+    plat.provision_with 'sudo dscl . -create /Users/test PrimaryGroupID 1000'
+    plat.provision_with 'sudo dscl . -create /Users/test NFSHomeDirectory /Users/test'
+    plat.provision_with 'sudo dscl . -passwd /Users/test password'
+    plat.provision_with 'sudo dscl . -merge /Groups/admin GroupMembership test'
+    plat.provision_with 'echo "test ALL=(ALL:ALL) NOPASSWD: ALL" > /etc/sudoers.d/username'
+    plat.provision_with 'mkdir -p /etc/homebrew'
+    plat.provision_with 'cd /etc/homebrew'
+    plat.provision_with 'su test -c \'echo | /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"\''
+    plat.provision_with 'sudo chown -R test:admin /Users/test/'
+    packages = %w(cmake)
+    plat.provision_with "su test -c '/usr/local/bin/brew install #{packages.join(' ')}'"
+    plat.vmpooler_template 'osx-1012-x86_64'
```

Comment:
This is still pointing to the 10.12 vmpooler template.

Reply:
Hey @ScottGarman ,
We have taken the same approach like in osx 10.13 that uses 10.12 template because when trying to build it with 10.14 it gives over 20 fatal errors shown below from a library named RAPIDJSON.
Do you think this approach is doable at this point?
```
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/internal/stack.h:38:5: error: macro expansion producing 'defined' has
      undefined behavior [-Werror,-Wexpansion-to-defined]
#if RAPIDJSON_HAS_CXX11_RVALUE_REFS
    ^
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/internal/../rapidjson.h:442:6: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
     ^
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/json_container/src/json_container.cc:4:
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/document.h:20:
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/reader.h:23:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/internal/stack.h:38:5: error: macro expansion producing 'defined' has
      undefined behavior [-Werror,-Wexpansion-to-defined]
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/internal/../rapidjson.h:442:34: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
                                 ^
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/json_container/src/json_container.cc:4:
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/document.h:20:
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/reader.h:23:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/internal/stack.h:60:5: error: macro expansion producing 'defined' has
      undefined behavior [-Werror,-Wexpansion-to-defined]
#if RAPIDJSON_HAS_CXX11_RVALUE_REFS
    ^
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/internal/../rapidjson.h:442:6: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
     ^
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/json_container/src/json_container.cc:4:
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/document.h:20:
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/reader.h:23:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/internal/stack.h:60:5: error: macro expansion producing 'defined' has
      undefined behavior [-Werror,-Wexpansion-to-defined]
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/internal/../rapidjson.h:442:34: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
                                 ^
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/json_container/src/json_container.cc:4:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/document.h:62:5: error: macro expansion producing 'defined' has undefined
      behavior [-Werror,-Wexpansion-to-defined]
#if RAPIDJSON_HAS_CXX11_RVALUE_REFS
    ^
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/error/../rapidjson.h:442:6: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
     ^
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/json_container/src/json_container.cc:4:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/document.h:62:5: error: macro expansion producing 'defined' has undefined
      behavior [-Werror,-Wexpansion-to-defined]
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/error/../rapidjson.h:442:34: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
                                 ^
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/json_container/src/json_container.cc:4:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/document.h:438:5: error: macro expansion producing 'defined' has undefined
      behavior [-Werror,-Wexpansion-to-defined]
#if RAPIDJSON_HAS_CXX11_RVALUE_REFS
    ^
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/error/../rapidjson.h:442:6: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
     ^
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/json_container/src/json_container.cc:4:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/document.h:438:5: error: macro expansion producing 'defined' has undefined
      behavior [-Werror,-Wexpansion-to-defined]
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/error/../rapidjson.h:442:34: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
                                 ^
/Library/Developer/CommandLineTools/usr/bin/make -f curl/CMakeFiles/leatherman_curl.dir/build.make curl/CMakeFiles/leatherman_curl.dir/build
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/json_container/src/json_container.cc:4:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/document.h:600:5: error: macro expansion producing 'defined' has undefined
      behavior [-Werror,-Wexpansion-to-defined]
#if RAPIDJSON_HAS_CXX11_RVALUE_REFS
    ^
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/error/../rapidjson.h:442:6: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
     ^
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/json_container/src/json_container.cc:4:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/document.h:600:5: error: macro expansion producing 'defined' has undefined
      behavior [-Werror,-Wexpansion-to-defined]
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/error/../rapidjson.h:442:34: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
                                 ^
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/json_container/src/json_container.cc:4:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/document.h:1042:5: error: macro expansion producing 'defined' has
      undefined behavior [-Werror,-Wexpansion-to-defined]
#if RAPIDJSON_HAS_CXX11_RVALUE_REFS
    ^
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/error/../rapidjson.h:442:6: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
     ^
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/json_container/src/json_container.cc:4:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/document.h:1042:5: error: macro expansion producing 'defined' has
      undefined behavior [-Werror,-Wexpansion-to-defined]
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/error/../rapidjson.h:442:34: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
                                 ^
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/json_container/src/json_container.cc:4:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/document.h:1303:5: error: macro expansion producing 'defined' has
      undefined behavior [-Werror,-Wexpansion-to-defined]
#if RAPIDJSON_HAS_CXX11_RVALUE_REFS
    ^
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/error/../rapidjson.h:442:6: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
     ^
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/json_container/src/json_container.cc:4:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/document.h:1303:5: error: macro expansion producing 'defined' has
      undefined behavior [-Werror,-Wexpansion-to-defined]
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/error/../rapidjson.h:442:34: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
                                 ^
/Library/Developer/CommandLineTools/usr/bin/make -f execution/CMakeFiles/leatherman_execution.dir/build.make execution/CMakeFiles/leatherman_execution.dir/build
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/json_container/src/json_container.cc:4:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/document.h:1738:5: error: macro expansion producing 'defined' has
      undefined behavior [-Werror,-Wexpansion-to-defined]
#if RAPIDJSON_HAS_CXX11_RVALUE_REFS
    ^
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/error/../rapidjson.h:442:6: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
     ^
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/json_container/src/json_container.cc:4:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/document.h:1738:5: error: macro expansion producing 'defined' has
      undefined behavior [-Werror,-Wexpansion-to-defined]
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/error/../rapidjson.h:442:34: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
                                 ^
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/json_container/src/json_container.cc:4:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/document.h:1757:5: error: macro expansion producing 'defined' has
      undefined behavior [-Werror,-Wexpansion-to-defined]
#if RAPIDJSON_HAS_CXX11_RVALUE_REFS
    ^
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/error/../rapidjson.h:442:6: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
     ^
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/json_container/src/json_container.cc:4:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/document.h:1757:5: error: macro expansion producing 'defined' has
      undefined behavior [-Werror,-Wexpansion-to-defined]
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/error/../rapidjson.h:442:34: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
                                 ^
In file included from /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/json_container/src/json_container.cc:5:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/stringbuffer.h:20:5: error: macro expansion producing 'defined' has
      undefined behavior [-Werror,-Wexpansion-to-defined]
#if RAPIDJSON_HAS_CXX11_RVALUE_REFS
    ^
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.2JBB66Gj/leatherman/src/rapidjson/include/rapidjson/error/../rapidjson.h:442:6: note: expanded from macro
      'RAPIDJSON_HAS_CXX11_RVALUE_REFS'
    (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__) && __GLIBCXX__ >= 20080306)
     ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
```

Reply:
Has there been a leatherman bug filed for this issue? If the 10.13 agent is passing tests and working properly, then this approach is probably fine. However this error could imply a library incompatibility that might cause problems later on.

@branan - could you chime in on this?

I'll be on PTO through the end of the year starting tomorrow, so if no other objections are raised, I'm :+1: on this.

Reply:
This is the same issue of rapidjson not playing nice with newer compilers that we noticed on newer Linuxes as well - it's not a system library compatibility issue, and should be fine to continue to build on 10.12 just like we do for the 10.13 DMGs.

I guess the question, then, is whether it makes sense to try to move to a single DMG that works on 10.12+, instead of building the same darn code three times >_<